kernel/aster-nix/src/fs/utils/channel.rs

```diff
@@ -21,15 +20,13 @@ pub struct Channel<T> {
 }
 
 impl<T> Channel<T> {
-    pub fn with_capacity(capacity: usize) -> Result<Self> {
-        Self::with_capacity_and_flags(capacity, StatusFlags::empty())
-    }
+    pub fn new(capacity: usize) -> Self {
```

Comment:
The original `Common::with_capacity_and_flags` checks `capacity` to ensure `capacity` is not zero, because `HeapRb::new` will panic if capacity is zero. I suggest documenting the panic case for the `new` method or add a `debug
_assert!(capacity!=0)`.

Reply:
This makes sense. I forgot to check the panic case for `HeapRb::new`. I've added documentation for `Channel::new`.